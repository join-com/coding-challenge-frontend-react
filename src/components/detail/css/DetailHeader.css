.DetailHeader__container {
  background-color: var(--section-background-color);
  display: flex;
  flex-wrap: wrap;
}

.DetailHeader__title {
  width: 100%;
  flex-basis: 100%;
}

.DetailHeader__back {
  margin: var(--section-margin) 0;
}
