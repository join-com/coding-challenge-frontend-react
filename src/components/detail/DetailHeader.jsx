import React from 'react';
import { Link } from 'react-router-dom';

import dataItemType from '../types/dataItem';

import I18n from '../../utils/I18n';

import './css/DetailHeader.css';

const DetailHeader = ({
  item, ...otherProps
}) => {
  if (item) {
    const { id } = item;

    return (
      <div className="DetailHeader">
        <div className="DetailHeader__container">
          <h2 className="DetailHeader__title">
            {I18n.t('en.data.title', { id })}
          </h2>
          <Link
            className="DetailHeader__back Button"
            to="/bikes"
          >
            {I18n.t('en.actions.back')}
          </Link>
        </div>
      </div>
    );
  }
  return null;
};

export default DetailHeader;

DetailHeader.propTypes = {
  item: dataItemType,
};

DetailHeader.defaultProps = {
  item: {},
};
