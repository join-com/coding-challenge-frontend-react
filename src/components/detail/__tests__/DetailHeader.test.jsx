import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import staticData from '../../../static/data.json';
import DetailHeader from '../DetailHeader';

const { incidents } = staticData;

const props = {
  item: incidents[0],
};

describe('DetailHeader', () => {
  it('should render the component correctly', () => {
    const tree = shallow(<DetailHeader {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should not render the component if no item', () => {
    const tree = shallow(<DetailHeader {...props} item={null} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
