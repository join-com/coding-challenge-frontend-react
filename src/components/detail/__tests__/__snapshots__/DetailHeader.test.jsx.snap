// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailHeader should not render the component if no item 1`] = `""`;

exports[`DetailHeader should render the component correctly 1`] = `
<div
  className="DetailHeader"
>
  <div
    className="DetailHeader__container"
  >
    <h2
      className="DetailHeader__title"
    >
      Stolen Bike NÂ° 94547
    </h2>
    <Link
      className="DetailHeader__back Button"
      replace={false}
      to="/bikes"
    >
      &lt; Back to list
    </Link>
  </div>
</div>
`;
